gpkit/geometric_program.py

```diff
@@ -103,14 +105,16 @@ def solve(self, solver=None, verbosity=1, *args, **kwargs):
                 posynomials : array of floats
                     Each posynomials's dual variable value at the solution.
         """
-        if solver is None:
+        def _default_solver():
             from . import settings
             if settings['installed_solvers']:
-                solver = settings['installed_solvers'][0]
+                return settings['installed_solvers'][0]
             else:
                 raise ValueError("No solver was given; perhaps gpkit was not"
                                  " properly installed, or found no solvers"
                                  " during the installation process.")
+        if solver is None:
+            solver = _default_solver()
```

Comment:
Just implemented that -- see 6afb5b0623276bfe0d036709c4769eba5615717d and please review